File: problems/p025.js
var ln10 = Math.log(10);

function log10(num) {
    return Math.log(num) / ln10;
}

var logPhi = log10((Math.sqrt(5) + 1) / 2);
var logSqrt5 = log10(5) / 2;

function digitsOfFibonacciNumber(i) {
    return Math.floor((i * logPhi) - logSqrt5) + 1;
}

var index = 0;
var digits;

do {
    digits = digitsOfFibonacciNumber(++index);
} while(digits < 1000);

console.log(index);

File: rust/p025.rs
fn digits_of_fibonacci(number: usize) -> usize {
    let log_phi = ((5f32.sqrt() + 1f32) / 2f32).log10();
    let log_sqrt5 = 5f32.log10() / 2f32;
    (((number as f32 * log_phi) - log_sqrt5).floor() + 1f32) as usize
}

fn main() {
    let result = (1..).into_iter()
        .find(|&x| digits_of_fibonacci(x) >= 1000)
        .unwrap();
    println!("{}", result);
}
